FirebaseMLModelDownloader/Sources/ModelDownloaderDeviceLogger.swift

```diff
@@ -0,0 +1,37 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import os
+
+/// On-device logger.
+class DeviceLogger {
+  /// Log event on device.
+  static func logEvent(level: OSLogType, category: OSLog, message: StaticString,
+                       messageCode: LoggerMessageCode) {
+    os_log(message, log: category, type: level)
+  }
+}
+
+/// Extension to categorize on-device logging.
+extension OSLog {
+  private static let subsystem: String = {
+    let bundleID = Bundle.main.bundleIdentifier ?? ""
+    return "com.google.firebaseml.\(bundleID)"
+  }()
+
+  /// List of logging categories.
+  static let modelDownload = OSLog(subsystem: subsystem, category: "model-download")
+  static let analytics = OSLog(subsystem: subsystem, category: "analytics")
+}
```

Comment:
Hmm, so other SDKs use [`FIRLogger`](https://github.com/firebase/firebase-ios-sdk/blob/c05801f1d127b1d360693fd7c2a8c75a0dea9374/FirebaseCore/Sources/Private/FIRLogger.h#L110-L119) but it's a private API within Core, and we likely won't be able to re-define these in an extension to make them visible since they're C definitions. 

This'll take some thinking of how we want to do this, we eventually want to switch to `os_log` for logging but this would be inconsistent with the rest of the SDKs.

Reply:
Ah yes I saw this, but I couldn't figure out how to use this in Swift. Is that possible at this point?